### PR TITLE
[5.1] Reconnect if lost connection during PDO instantiation

### PR DESCRIPTION
**Pull request by monque**
#10146

### CODE REVIEW

**Comment by monque** [comment]
#10146

**Comment by monque** [comment]
Already fix styleci issue.

This issue is tightly-related with specified database implement. Perhaps it can be tested by a mocked PDO connection, and it will only throw a `Exception('Some database gone away');`, but it doesn't make any sense.

Do you have any suggestion ?

File: src/Illuminate/Database/Connectors/Connector.php [hunk 1]
```diff
@@ -70,4 +84,25 @@ public function setDefaultOptions(array $options)
     {
         $this->options = $options;
     }
+
+    /**
+     * Handle a exception that occurred during connect execution.
+     *
+     * @param  \Exception  $e
+     * @param  string  $dsn
+     * @param  string  $uesrname
```
Comment (vlakoff) [inline src/Illuminate/Database/Connectors/Connector.php]: typo "username"

**Comment by taylorotwell** [comment]
Going to hold off on this pending more reports from the community.

**Comment by GrahamCampbell** [comment]
@taylorotwell https://github.com/laravel/framework/issues/10204

**Comment by GrahamCampbell** [comment]
Please rebase this.

**Comment by GrahamCampbell** [comment]
Please rebase this.

**Pull request merged**